Gemfile

```diff
@@ -81,3 +83,7 @@ group :development do
   # Letter Opener for previewing emails in the browser https://github.com/fgrehm/letter_opener_web
   gem 'letter_opener_web'
 end
+
+group :test do
+  gem 'mocha'
```

Comment:
I did not add tests for the js part, probably should too. That gem however is not the `js` mocking library `mocha`, rather the ruby library of the same name (maybe you realised that and the comment was just about the tests?).